HPXMLtoOpenStudio/resources/schedules.rb

```diff
@@ -1641,6 +1642,17 @@ def create_column_values_from_periods(col_name, periods)
     end
   end
 
+  def expand_schedules
+    # Expand schedules with fewer elements such that all the schedules have the same number of elements
+    num_hrs_in_year = Constants.NumHoursInYear(@year)
+    n_ts_per_hr = @tmp_schedules.map { |_k, v| v.size / num_hrs_in_year }.uniq.max
+    @tmp_schedules.each do |col, values|
+      if values.size / num_hrs_in_year < n_ts_per_hr
+        @tmp_schedules[col] = values.map { |v| [v] * n_ts_per_hr }.flatten
+      end
+    end
```

Comment:
Admittedly I haven't tested it, but this doesn't seem right. Why are you using `n_ts_per_hr` for all schedule columns regardless of how many elements they have? If there are three columns, one with 50k elements, one with 25k elements, and one with 5k elements, why would the 25k and 5k columns both get expanded by the same number of elements? Also, I don't see why `num_hrs_in_year` is needed, it cancels out in the `if` comparison.

Shouldn't it be something like this?
```suggestion
    max_size = @tmp_schedules.map { |_k, v| v.size }.uniq.max
    @tmp_schedules.each do |col, values|
      if values.size < max_size
        @tmp_schedules[col] = values.map { |v| [v] * max_size / values.size }.flatten
      end
    end
```